Mancala.hs

```diff
@@ -31,36 +31,92 @@ mancala_start = State (side_start, side_start) (actions_start, actions_start)
 --mancala move (State ((my_pockets, my_store), (other_pockets, other_store)) (my_actions, other_actions))
 -- - move pieces
 
+--precondition: index >= 0
 get_val :: Board -> Int -> Int
 get_val board index = (to_list board) !! index
--- TODO: tests
-
+{-
+*Mancala> myBoard = ([1..7], [8..14])
+*Mancala> myBoard
+([1,2,3,4,5,6,7],[8,9,10,11,12,13,14])
+*Mancala> get_val myBoard 0
+1
+*Mancala> get_val myBoard 13
+14
+*Mancala> get_val myBoard 14
+*** Exception: Prelude.!!: index too large
+*Mancala> get_val myBoard 8
+9
+-}
 set_val :: Board -> Int -> Int -> Board
 set_val (my_side, other_side) index value = to_board new_l other_store len
   where
     len = length $ get_pockets my_side
     other_store = get_store other_side
     l = to_list (my_side, other_side)
     new_l = (take index l) ++ (value : drop (index+1) l)
----- TODO: tests
-
+{-
+*Mancala> myBoard = ([1..7], [8..14])
+*Mancala> myBoard
+([1,2,3,4,5,6,7],[8,9,10,11,12,13,14])
+*Mancala> set_val myBoard 0 100
+([100,2,3,4,5,6,7],[8,9,10,11,12,13,14])
+*Mancala> set_val myBoard 12 100
+([1,2,3,4,5,6,7],[8,9,10,11,12,100,14]) -- if used on last index, will add a value
+*Mancala> set_val myBoard 2 100
+([1,2,100,4,5,6,7],[8,9,10,11,12,13,14])
+-}
 get_pockets :: Side -> [Int]
 get_pockets s = init s
-
+{-
+*Mancala> mySide = [1..7]
+*Mancala> mySide
+[1,2,3,4,5,6,7]
+*Mancala> get_pockets mySide
+[1,2,3,4,5,6]
+-}
 get_store :: Side -> Int
 get_store s = last s
-
+{-
+*Mancala> mySide = [1..7]
+*Mancala> mySide
+[1,2,3,4,5,6,7]
+*Mancala> get_store mySide
+7
+-}
 to_list :: Board -> [Int]
 to_list (my_side, other_side) = my_side ++ (get_pockets other_side) 
--- TODO: tests
+{-
+*Mancala> to_list myBoard
+[1,2,3,4,5,6,7,8,9,10,11,12,13]
+*Mancala> myBoard = ([1..7],[8..14])
+*Mancala> myBoard
+([1,2,3,4,5,6,7],[8,9,10,11,12,13,14])
+*Mancala> to_list myBoard
+[1,2,3,4,5,6,7,8,9,10,11,12,13]
+-}
 
 to_board :: [Int] -> Int -> Int -> Board 
 to_board l other_store len = 
```

Comment:
I should definitely add documentations for this function haha.
l is the result of a to_list call
other_store is the opponent's store/mancala
len is the number of pockets + 1 (mancala)